### PR TITLE
`/matrix/*` tests

### PR DESCRIPTION
**Pull request by FiIipson**

### CODE REVIEW

**Review by Pankin610 — APPROVED** [review]
Nice code bro, great test names and code style.

**Comment by FiIipson** [comment]
> Nice code bro, great test names and code style.

Thanks for the review :muscle:

**Comment by FiIipson** [comment]
**Please do not merge this branch by the end of day** still updating new stuff.

**Review by Pankin610 — CHANGES_REQUESTED** [review]

File: java/src/test/java/linalg/matrix/COOMatrixTest.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    matrix.ForEachEntry(entry -> map.put(entry.Coord(), entry.Value()));
+
+    COOMatrix coo_matrix = new COOMatrix(map, test_size, test_size);
+    Assertions.assertEquals(coo_matrix.Rows(), test_size);
+  }
+
+  @Test
+  void RowsWithEmptyMap() {
+    for (int i = 0; i < test_size; i++)
+      Assertions.assertEquals(i, new COOMatrix(new HashMap<>(), i, i + new Random().nextInt(test_size)).Rows());
```
[the remark this answers is not in the document]
Comment (FiIipson) [reply java/src/test/java/linalg/matrix/COOMatrixTest.java:33]: Indeed, forgot about that rule - usually my IDE does it for me

**Pull request merged**